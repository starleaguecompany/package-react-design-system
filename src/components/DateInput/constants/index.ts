export const DAYS_OF_WEEK = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']
export const MONTHS = [
  'Январь',
  'Февраль',
  'Март',
  'Апрель',
  'Май',
  'Июнь',
  'Июль',
  'Август',
  'Сентябрь',
  'Октябрь',
  'Ноябрь',
  'Декабрь',
]
