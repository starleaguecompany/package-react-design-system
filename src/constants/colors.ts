export const COLORS = {
  GRAY: 'gray',
  GREEN: 'green',
  BLUE: 'blue',
  ORANGE: 'orange',
  VIOLET: 'violet',
  RED: 'red',
} as const
