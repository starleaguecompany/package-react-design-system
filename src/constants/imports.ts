export const reactDSImportPath = '@starleaguecompany/react-design-system';
