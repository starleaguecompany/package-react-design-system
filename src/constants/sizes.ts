export const CONTENT_SIZES = {
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
} as const;

export const CONTAINER_SIZES = {
  S36: 36,
  S44: 44,
  S52: 52,
  S60: 60,
} as const;

export const FONT_SIZES = {
  S10: 10,
  S12: 12,
  S14: 14,
  S16: 16,
  S20: 20,
} as const;
