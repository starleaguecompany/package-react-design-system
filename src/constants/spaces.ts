export const SPACE_SIZES = {
  S0: 0,
  S2: 2,
  S4: 4,
  S8: 8,
  S12: 12,
  S16: 16,
  S20: 20,
  S24: 24,
  S28: 28,
  S32: 32,
  S36: 36,
  S40: 40,
} as const;
