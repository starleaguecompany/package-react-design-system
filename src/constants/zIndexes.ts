export const Z_INDEXES = {
  DROPDOWN: 1060,
  STICKY: 1020,
  FIXED: 1030,
  MODAL_BACKDROP: 1040,
  MODAL: 1050,
  TOOLTIP: 1060,
} as const;
