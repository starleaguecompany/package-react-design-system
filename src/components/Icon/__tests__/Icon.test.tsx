import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { CONTENT_SIZES } from '../../../constants/sizes';
import { Menu } from '@starleaguecompany/react-icons';
import { Icon } from '../index';

describe('Icon', () => {
  test('should render correctly without prop', () => {
    render(<Icon />);

    expect(screen.getByTestId('Icon')).toMatchSnapshot();
    expect(screen.getByTestId('Icon')).toHaveClass('container size-20 shape-round', { exact: true });
  });

  test('should render correctly with attributes', () => {
    render(<Icon id="test-id" className="test-class" icon={<Menu />} />);

    expect(screen.getByTestId('Icon')).toMatchSnapshot();
    expect(screen.getByTestId('Icon')).toHaveClass('test-class');
    expect(screen.getByTestId('Icon')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly with icon', () => {
    render(<Icon icon={<Menu />} />);

    expect(screen.getByTestId('Icon')).toMatchSnapshot();
    expect(screen.getByTestId('Icon')).toContainElement(screen.getByTestId('Menu'));
    expect(screen.getByTestId('Menu')).toBeVisible();
  });

  test('should render correctly with container', () => {
    render(
      <Icon>
        <Menu />
      </Icon>
    );

    expect(screen.getByTestId('Icon')).toMatchSnapshot();
    expect(screen.getByTestId('Menu')).toBeVisible();
  });

  test('should render correctly with strong prop', () => {
    render(<Icon strong icon={<Menu />} />);

    expect(screen.getByTestId('Icon')).toMatchSnapshot();
    expect(screen.getByTestId('Icon')).toHaveClass('strong');
  });

  test('should render correctly with color prop', () => {
    render(<Icon color="blue" icon={<Menu />} />);

    expect(screen.getByTestId('Icon')).toMatchSnapshot();
    expect(screen.getByTestId('Icon')).toHaveClass('color-blue');
  });

  test('should render correctly with shape prop', () => {
    render(<Icon shape="circle" icon={<Menu />} />);

    expect(screen.getByTestId('Icon')).toMatchSnapshot();
    expect(screen.getByTestId('Icon')).toHaveClass('shape-circle');
  });

  test('should render correctly with size prop', () => {
    render(<Icon size={CONTENT_SIZES.S16} icon={<Menu />} />);

    expect(screen.getByTestId('Icon')).toMatchSnapshot();
    expect(screen.getByTestId('Icon')).toHaveClass('size-16');
  });
});
