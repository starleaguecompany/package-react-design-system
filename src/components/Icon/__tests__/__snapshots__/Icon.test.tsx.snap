// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render correctly with attributes 1`] = `
<span
  class="test-class container size-20 shape-round"
  data-qa="Icon"
  id="test-id"
>
  <svg
    data-qa="Menu"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 5h12M4 10h12M4 15h12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</span>
`;

exports[`Icon should render correctly with color prop 1`] = `
<span
  class="container size-20 color-blue shape-round"
  data-qa="Icon"
>
  <svg
    data-qa="Menu"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 5h12M4 10h12M4 15h12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</span>
`;

exports[`Icon should render correctly with container 1`] = `
<span
  class="container size-20 asContainer shape-round"
  data-qa="Icon"
>
  <svg
    data-qa="Menu"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 5h12M4 10h12M4 15h12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</span>
`;

exports[`Icon should render correctly with icon 1`] = `
<span
  class="container size-20 shape-round"
  data-qa="Icon"
>
  <svg
    data-qa="Menu"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 5h12M4 10h12M4 15h12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</span>
`;

exports[`Icon should render correctly with shape prop 1`] = `
<span
  class="container size-20 shape-circle"
  data-qa="Icon"
>
  <svg
    data-qa="Menu"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 5h12M4 10h12M4 15h12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</span>
`;

exports[`Icon should render correctly with size prop 1`] = `
<span
  class="container size-16 shape-round"
  data-qa="Icon"
>
  <svg
    data-qa="Menu"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 5h12M4 10h12M4 15h12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</span>
`;

exports[`Icon should render correctly with strong prop 1`] = `
<span
  class="container strong size-20 shape-round"
  data-qa="Icon"
>
  <svg
    data-qa="Menu"
    fill="none"
    viewBox="0 0 20 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M4 5h12M4 10h12M4 15h12"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
    />
  </svg>
</span>
`;

exports[`Icon should render correctly without prop 1`] = `
<span
  class="container size-20 shape-round"
  data-qa="Icon"
/>
`;
